server/models/estimationFeatureModel.js

```diff
@@ -276,5 +277,75 @@ estimationFeatureSchema.statics.updateFeatureByEstimator = async (featureInput,
     return await estimationFeature.save()
 }
 
+estimationFeatureSchema.statics.updateFeatureByNegotiator = async (featureInput, negotiator) => {
+    validate(featureInput, estimationNegotiatorUpdateFeatureStruct)
+
+    if (!negotiator || !userHasRole(negotiator, SC.ROLE_NEGOTIATOR))
+        throw new AppError('Not an negotiator', EC.INVALID_USER, EC.HTTP_BAD_REQUEST)
+
+    let estimationFeature = await EstimationFeatureModel.findById(featureInput._id)
+    if (!estimationFeature)
+        throw new AppError('Estimation feature not found', EC.NOT_FOUND, EC.HTTP_BAD_REQUEST)
+
+    let estimation = await EstimationModel.findOne({"_id": estimationFeature.estimation._id})
+    if (!estimation)
+        throw new AppError('Estimation not found', EC.NOT_FOUND, EC.HTTP_BAD_REQUEST)
+
+    if (!_.includes([SC.STATUS_INITIATED, SC.STATUS_REVIEW_REQUESTED], estimation.status))
+        throw new AppError("Estimation has status as [" + estimation.status + "]. Negotiator can only add feature into those estimations where status is in [" + SC.STATUS_INITIATED + ", " + SC.STATUS_REVIEW_REQUESTED + "]", EC.INVALID_OPERATION, EC.HTTP_BAD_REQUEST)
+
+    if (!estimation.negotiator._id == negotiator._id)
+        throw new AppError('You are not negotiator of this estimation', EC.INVALID_USER, EC.HTTP_FORBIDDEN)
+
+    /**
+     * Check to see if this feature updated by negotiator or not
+     */
+    if (estimationFeature.owner == SC.OWNER_NEGOTIATOR && !estimationFeature.addedInThisIteration && !estimationFeature.estimator.changeRequested && !estimationFeature.estimator.changeGranted) {
```

Comment:
Negotiator can edit a feature any time, it does not need permission of estimator to do that